nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -1300,6 +1270,7 @@ define(
                     }
                 })();
                 fsm.newState(newFsmState);
+                ui.setContent('run-control-panel.execMessage', newExecMessage);
```

Comment:
moved the setting of these messages to within the functions that call them, instead of making them part of the FSM stuff => lessening dependence on the FSM.